File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = {};

    for (let i = 0; i < nums.length; i++) {
        let diff = target - nums[i];
        if (map[diff] !== undefined && map[diff] !== i) {
            return [map[diff], i];
        }
        map[nums[i]] = i;
    }
    return [];
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    idxMap := make(map[int]int)
    for i, num := range nums {
        diff := target - num
        if index, found := idxMap[diff]; found {
            return []int{index, i}
        }
        idxMap[num] = i
    }
    return nil
}